packages-exp/auth-exp/src/core/util/assert.ts

```diff
@@ -15,29 +15,65 @@
  * limitations under the License.
  */
 
-import { AUTH_ERROR_FACTORY, AuthErrorCode } from '../errors';
+import { AuthErrorCode, AUTH_ERROR_FACTORY } from '../errors';
+import { logError } from './log';
 
-export function assert<T>(
-  expression: T | null | undefined,
+/**
+ * Unconditionally fails, throwing a developer facing INTERNAL_ERROR
+ *
+ * @param appName App name for tagging the error
+ * @throws FirebaseError
+ */
+export function fail(appName: string, errorCode: AuthErrorCode): never {
+  throw AUTH_ERROR_FACTORY.create(errorCode, { appName });
+}
+
+/**
+ * Verifies the given condition and fails if false, throwing a developer facing error
+ *
+ * @param assertion
+ * @param appName
+ */
+export function assert(
+  assertion: unknown,
   appName: string,
-  code = AuthErrorCode.INTERNAL_ERROR
-): T {
-  if (!expression) {
-    throw AUTH_ERROR_FACTORY.create(code, { appName });
+  errorCode: AuthErrorCode = AuthErrorCode.INTERNAL_ERROR
```

Comment:
nit: no need to type this since it has a default value: go/tsstyle#type-inference